part-3/part-3.md

```diff
@@ -152,252 +153,289 @@ const source = from([1, 2, 3]).pipe(
 source.subscribe((value) => console.log(value));
 ```
 
-Как видно из примера, мы заменили **filter()** на оператор **skip()**, что сделало наш код компактнее и понятнее. Теперь наш код смотрится более декларативно.
+We replaced **filter()** with **skip()**, which made our code more readable and understandable. Now our code looks more declarative.
 
-Давайте теперь возьмем только первые несколько значений из потока. Для этого нам понадобятся операторы **take()** и **first()**.
+Now let's do the opposite, that is, take only the first several values from a stream, and ignore the rest. For that purpose, we will need operators **take()** and **first()**.
 
-**take()** берет указанное количество значений из потока, и завершает поток.
+**take()** takes the first several values in the stream (provided in the argument), and completes the stream.
 
-**first()** берет только первое значение из потока, и завершает поток. Обратите внимание, что если поток не вернет значение, то будет выброшено исключение **EmptyError**.
+**first()** takes only the very first value from the stream and complete it. Notice that if the stream completes itself before emitting any value, an `EmptyError` will be thrown, so **first()** is fundamentally different from **take(1)**.
 
 ```ts
-import { from } from "rxjs";
+import { from } from 'rxjs';
 
-import { take, first } from "rxjs/operators";
+import { take, first } from 'rxjs/operators';
 
 const source = from([1, 2, 3]);
 
 const sourceTakeTwo = source.pipe(
-  // Возьмет первые два значения и завершит поток
+  // will take the first two values from the stream and complete it 
 
-  take(2)
+  take(2),
 );
 
-// Выведет в консоль:
+// Will log:
 
 //. 1
 
 //. 2
 
-sourceTakeTwo.subscribe((value) => console.log(value));
+sourceTakeTwo.subscribe(value => console.log(value));
 
 const sourceTakeFirst = source.pipe(
-  // Возьмет первое значение и завершит поток
+  // Will take the ver first value and complete the stream
 
-  first()
+  first(),
 );
 
-// Выведет в консоль:
+// Will log:
 
 //. 1
 
-sourceTakeFirst.subscribe((value) => console.log(value));
+sourceTakeFirst.subscribe(value => console.log(value));
 ```
 
-Отлично! Теперь мы умеем брать несколько первых или несколько последних значений из потока. Давайте посмотрим, как еще можно фильтровать значения с помощью операторов **RxJS**.
+Great!Now we know how to take the first values from a stream. Let's see how else can we filter values using **RxJS** operators.
+
+## **distinct()**
+
+Sometimes we need to only handle unique values, or in other words, ignore duplicates. For that purposes we can utilize the **distinct** operator. It will allow only those values that have not been emitted yet.
```

Comment:
```suggestion
Sometimes we only want to operate on unique values, or in other words, ignore duplicates. For that purposes we can utilize the **distinct** operator. It will only pass values that have not been emitted yet.
```